netket/driver/vmc.py

```diff
@@ -101,12 +119,14 @@ def _forward_and_backward(self):
         # Compute the local energy estimator and average Energy
         self._loss_stats, self._loss_grad = self.state.expect_and_grad(self._ham)
 
-        if self.sr is not None:
-            self._S = self.state.quantum_geometric_tensor(self.sr)
+        if self.preconditioner is not None:
+            self._S = self.preconditioner.object(self.state)
 
             # use the previous solution as an initial guess to speed up the solution of the linear system
-            x0 = self._dp if self.sr_restart is False else None
-            self._dp, self._sr_info = self._S.solve(self._loss_grad, x0=x0)
+            x0 = self._dp if self.preconditioner_restart is False else None
+            self._dp, self._sr_info = self._S.solve(
+                self.preconditioner.solver, self._loss_grad, x0=x0
+            )
```

Comment:
I made sr a NamedTuple with two fields, `preconditioned` and `object`. 
it will behave as a tuple but you can also address the two fields by name.

I don't know if we like that approach. It seems cleaner but... I dunno.
what do you think @femtobit ?

Reply:
I think using a named tuple is pretty reasonable.

(btw, I haven't had time to catch up with the rest of the QGT discussion until now, but I'll look at the rest of the changes as soon as i have a quiet moment.)